### DISCUSSION TITLE
Documenting & pruning GitHub labels

### DISCUSSION THREAD

**Discussion by evan-goode** [rpm-software-management/dnf5#821] [General]
Hi, I would like to start a discussion about the labels we use on this repository. They are listed here: https://github.com/rpm-software-management/dnf5/issues/labels. It would be nice to go through them and (1) document what they are for and when they should be set and (2) determine whether any old/redundant ones are still necessary. I went ahead and added descriptions for a few, let me know whether these seem correct.

I propose removing the following unused or inconsistently-used labels:

- `dnf5`: probably vestigial from another repo, everything is DNF 5 here :)
- `deferred`: does not seem used?
- `wontfix`: does not seem used?
- `ready for review` & `work in progress`: we should use Draft PRs for this instead, it's easy to forget to set `ready for review` so it doesn't have much meaning, I would rather have "ready for review" be the default state of a non-draft PR, and ditch the label.

I also am not sure what purpose the `changes requested` label has, it seems GitHub automatically makes it visible on the issue when a reviewer has requested changes, and when the reviewer has marked those changes as addressed. But maybe I am missing something.

cc @mcurlej @j-mracek

**Comment by kontura** [comment]
I like this proposal.
Mainly I think dropping the `ready for review`, `work in progress` and `changes requested` is a good idea.

- Converting a PR back to draft is better than `work in progress` because it is harder to overlook and such PR cannot be easily merged.
- Requesting changes through the review process is also better than `changes requested` label because it prompts for a message automatically. The `changes requested` label by it self is not very useful it should go together with a message.

**Comment by j-mracek** [comment]
I am proposing to drop:

- `deferred` - not sure about purpose
- `dnf5` - inherited from `libdnf`
- `ready for review` - not required anymore
- `changes requested` - sound like redundant with github review process.

I am suggesting to keep:

- `wontfix` - for issues it is nice to state reason why issue was closed

I am not sure about `work in progress`  - sometimes PR requires significant redesign and then it is nice to somehow state the status. Blocked can be used for the purpose, but might be not the best choice.

**Comment by ppisar** [comment]
I overall agree. My observations:

* deferred - Can be replaced with wontfix. I actually used as a temporary, soft wontfix: The request made a sense in the long term, but not with current state of the environment.
* dnf5 - Remove it.
* wontfix - GitHub has "Close as completed" and "Close as not planned". I don't know how they are represented to the users. The latter one si similar to wontfix.
* bug - First I thought it as a complement to RFE. But the description highlights a downstream origin. Do we really use it and does it make sense in a form of a label? I don't find this label without a link to a downstream issue tracker helpfull. Then the only what remains is an implicit higher priority because it  probably affects a lot of users. But we have priority labels for that.

**Comment by evan-goode** [comment]
It sounds like the consensus is to drop "deferred" and "dnf5", and to use GitHub's features instead of "ready for review" and "changes requested". I opened https://github.com/rpm-software-management/dnf5/pull/905 to update our CONTRIBUTING.md and if that's merged I will remove those labels.

 I think it's fine to keep "work in progress" perhaps to signify a PR is a bigger, long-term project.

@ppisar Regarding `bug`, I agree, it's not used consistently, I'm not sure we need it. Maybe @mcurlej has an opinion?
